client/node.js

```diff
@@ -86,6 +87,10 @@ module.exports = client(function node (request) {
       reject(response)
     }
 
+    if (request.agent) {
```

Comment:
It's not necessary to add a new property to the request to set the agent. You can currently define the agent via `request.mixin.agent`.